src/osemgrep/cli_scan/Output.ml

```diff
@@ -175,9 +166,29 @@ let output_result (conf : Scan_CLI.conf) (profiler : Profiler.t)
       (* TODO(dinosaure): [false] depends on the output formatter. Currently,
          we just have the JSON output. *)
     in
-    Nosemgrep.process_ignores ~keep_ignored ~strict:conf.Scan_CLI.strict
-      cli_output
+    cli_output
+    |> Nosemgrep.process_ignores ~keep_ignored ~strict:conf.Scan_CLI.strict
+    |> fun results ->
+    {
+      results with
+      results = Cli_json_output.index_match_based_ids results.results;
+    }
   in
+  cli_output
+
+(* python: mix of output.OutputSettings(), output.OutputHandler(), and
+ * output.output() all at once.
+ * TODO: take a more precise conf than Scan_CLI.conf at some point
+ *)
+let output_result (conf : Scan_CLI.conf) (profiler : Profiler.t)
+    (res : Core_runner.result) : Out.cli_output =
+  (* In theory, we should build the JSON CLI output only for the
+   * Json conf.output_format, but cli_output contains lots of data-structures
+   * that are useful for the other formats (e.g., Vim, Emacs), so we build
+   * it here.
+   *)
+  let cli_output = preprocess_result conf res in
+  (* TOPORT? output.output() *)
   let cli_output = Profiler.record profiler ~name:"ignores_times" cli_output in
```

Comment:
why not removing this unit argument in preprocess_result and instead just do here
let cli_output = Profile.record profile ... (fun () ->
    preprocess_result conf res)
in

Reply:
Duh, that makes a lot of sense, will change it